Initial 1.4.0 release notes (#5248)
istio/istio.io: content/en/news/2019/announcing-1.4/change-notes/index.md

weight: 10
---

## Traffic management

> - **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).

Author: I am not quite sure what you mean. https://preliminary.istio.io/docs/concepts/traffic-management/ does not have anything about mirroring. We already have a doc that explains this, and I linked to it. Its not really a "different type" of mirroring, its just sending only x% of traffic instead of 100%. Would it make sense to just rephrase this to "Added support for mirroring a percentage of traffic"?

> - **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).

Reviewer: Yes that would be enough clarity.
The correct link for the mirroring task is this one: `https://istio.io/docs/tasks/traffic-management/mirroring/`

> - **Improved** the Envoy sidecar to exit upon crashing, giving better visibility of health.

Author: Not really.. previously it was completely obscured, now its just standard kubernetes. if it gets OOM killed kubernetes will mark it as OOMKilled. If it is crashing it will be marked as CrashLoopBackoff. Its both self explanatory and part of basic kubernetes docs, so I don't think we need to specifically document it on our tasks

- **Improved** Pilot to skip sending redundant configuration to Envoy when no changes are required.

## Security

> 

Reviewer 2: Can we add:
```
- **Added** v1beta1 authorization policy for access control which deprecates the v1alpha1 RBAC policy.
```

Author: @yangminzhu definitely needs to be added at some point, I was waiting for https://github.com/istio/istio.io/issues/5289 before adding this. Unlike previous release notes PRs I am trying to split this up and just get the notes that are ready in asap and we can continue to add others as they are ready

Reviewer 2: Sounds good, we can update this later when other authz-v2 docs are merged.

> - **Improved**Citadel to periodically check and rotate the expired root certificate when running in self-sign CA mode.

Reviewer: ```suggestion
- **Improved** Citadel to periodically check and rotate the expired root certificate when running in self-sign CA mode.
```

## Telemetry

- **Added** support for configuring [stat patterns](/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig) for Envoy stats.
- **Added** prefix to Envoy HTTP stats to specify traffic direction (`inbound` or `outbound`).

## Configuration management

- **Added** many additional validation checks to [`istioctl analyze`](/docs/ops/diagnostic-tools/istioctl-analyze/).
- **Added** experimental support for validation message in Istio [resource statuses](/docs/ops/diagnostic-tools/istioctl-analyze/#enabling-validation-messages-for-resource-status).

> - **Added** OpenAPI v3 schema validation to CRDs. Please see the [Upgrade Notes](/news/2019/announcing-1.4/upgrade-notes) for details.

Author: https://github.com/istio/istio.io/pull/5174 (already merged)

> ## Installation

Author: Yes, I added only the portion of release notes that were ready. I thought it would be easier to split this rather than have one mega-pr

> ## Installation

Reviewer 3: - **Added** [operator controller](/docs/ops/setup/standalone-operator/) for dynamic updates to Istio installation based on a user CustomResourceDefinition.

- **Removed** the `proxy_init` Docker image. Instead, the `istio-init` container reuses the `proxyv2` image.
- **Updated** the base image to `ubunutu:bionic`.

## `istioctl`

> 

Reviewer 4: Please also add one line for upgrade command:
```
- **Added** the experimental sub-command [`istioctl experimental upgrade`](docs/setup/upgrade/istioctl-upgrade/) to performs an in-place upgrade of the Istio control plane with eligibility checks.
```

> - **Added** [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to wait until an Istio configuration has been pushed to all Envoy sidecars.

Reviewer: ```suggestion
- **Added** the experimental sub-command [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to have Istio wait until it has pushed a configuration to all Envoy sidecars.
```

> - **Added** the experimental sub-command [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to have Istio wait until it has pushed a configuration to all Envoy sidecars.

Reviewer 5: ```suggestion
- **Added** the experimental [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) sub-command to have Istio wait until it has pushed a configuration to all Envoy sidecars.
```

> - **Added** [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) to help manage Istio across multiple clusters.

Reviewer: ```suggestion
- **Added** the experimental [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) sub-command to help manage Istio across multiple clusters.
```

> - **Added** [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) to manage webhook configuration.

Reviewer: ```suggestion
- **Added** the experimental [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) sub-command to manage webhook configuration.
```

> - **Added** [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) to manage webhook configuration.

Reviewer 3: - **Added** [`istioctl experimental upgrade`](/docs/reference/commands/istioctl/#istioctl-experimental-upgrade) to upgrade Istio installed with `istioctl manifest` commands.

> - **Improved** [`istioctl version`](/docs/reference/commands/istioctl/#istioctl-version) to show Envoy proxy versions.

Reviewer: ```suggestion
- **Improved** the [`istioctl version`](/docs/reference/commands/istioctl/#istioctl-version) sub-command. It now shows the Envoy proxy versions.
```

## Miscellaneous

> - **Replaced** daily builds of Istio with builds for [each commit](https://github.com/istio/istio/wiki/Dev%20Builds).

Reviewer 5: This doesn't belong in customer-facing release notes.

Author: I am not sure I agree with this. I have had a number of users ask where they can find these builds. It's at the bottom of the notes, is it harmful to have this here?

Reviewer 5: It's not harmful, but it goes against the general direction we're trying to follow of being customer-focused rather than developer/hacker-focused.
